### PR TITLE
(dev/core#2301) Profiles -  Add Drupal 8+ specific check and extract to CRM_Utils_System

### PR DESCRIPTION
**Pull request by herbdool**
Fixes https://lab.civicrm.org/dev/core/-/issues/2301

### CODE REVIEW

**Comment by seamuslee001** [comment]
Looks good to me do you agree @demeritcowboy ?

**Comment by herbdool** [comment]
Says it failed on an asset builder test? Seems unrelated

**Comment by seamuslee001** [comment]
yen that is a known failure at the moment

**Comment by demeritcowboy** [comment]
Also seems to be general mysql fails on several PRs right now.

I can review tomorrow.

File: CRM/Utils/System/Base.php [hunk 1]
```diff
@@ -1114,4 +1114,13 @@ public function checkCleanurls() {
     return [];
   }
 
+  /**
+   * Suppress profile form errors
+   *
+   * @return bool
+   */
+  public function suppressProfileFormErrors() {
```
Comment (seamuslee001) [inline CRM/Utils/System/Base.php:1122]: wonder if its useful to add a `:bool` type hint onto these functions

Comment (herbdool) [reply CRM/Utils/System/Base.php]: sure

**Comment by demeritcowboy** [comment]
Thanks @herbdool. This works to make it more like drupal 7.

There are still several problems with the form, such as not validating properly because it uses title instead of name to find the profile, and so now instead of an error it silently says "updated successfully" even when it didn't, but we can treat separately.

**Comment by herbdool** [comment]
@demeritcowboy yup, I noticed. At least it'll be easier to untangle the other problems after this too. And a bonus, this will now work in Backdrop as well.

**Pull request merged**